src/services/dexes/sovryn/SovrynPriceFetcher.ts

```diff
@@ -0,0 +1,20 @@
+import {PricesResponse, PairRequest, SovrynHelperBase} from '../../../blockchains/evm/contracts/SovrynHelper.js';
+import {FeedFetcherInterface} from 'src/types/fetchers.js';
+
+export class SovrynPriceFetcher implements FeedFetcherInterface {
+  sovrynConnection!: SovrynHelperBase;
+
+  constructor(sovrynConnection: SovrynHelperBase) {
+    this.sovrynConnection = sovrynConnection;
+  }
+
+  async getPrices(pairs: PairRequest[]): Promise<PricesResponse> {
+    const prices = await this.sovrynConnection.getPrices(pairs);
+    return prices;
+  }
+
+  async apply(pairs: PairRequest[]): Promise<number[]> {
+    const prices = this.getPrices(pairs);
+    return (await prices).prices.map((price) => price.price.toNumber());
```

Comment:
this will not work for big numbers, please add a unit test where you will use different prices. Use helper with different `amountDecimals` from 0 to 18 to generate real data, and then use this data to create unit test.

Reply:
is this addressed?